File: 202-happy-number/202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        hashSet = set()
        while((n != 1) and (n not in hashSet)) :
            hashSet.add(n)
            temp = 0
            while(n > 0) : 
                temp += (n%10)*(n%10)
                n = n//10
            
            n = temp
            
        return n == 1
            

File: 202-happy-number/202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> hset = new HashSet<>();
        
        while(n != 1 && !hset.contains(n)){
            int temp = 0;
            hset.add(n);
            while(n > 0){
                temp += (n%10)*(n%10);
                n /= 10;
            }
            n = temp;
        }
        
        return n == 1;
    }
}